core/tags.go

```diff
@@ -0,0 +1,129 @@
+package core
+
+import (
+	"context"
+	"net/url"
+
+	"github.com/augurysys/timestamp"
+	"github.com/krystal/go-katapult"
+)
+
+type Tag struct {
+	ID        string               `json:"id,omitempty"`
+	Name      string               `json:"name,omitempty"`
+	Color     string               `json:"color,omitempty"`
+	CreatedAt *timestamp.Timestamp `json:"created_at,omitempty"`
+}
+
+type TagsClient struct {
+	client   RequestMaker
+	basePath *url.URL
+}
+
+func NewTagsClient(rm RequestMaker) *TagsClient {
+	return &TagsClient{
+		client:   rm,
+		basePath: &url.URL{Path: "/core/v1/"},
+	}
+}
+
+type tagsResponseBody struct {
+	Pagination *katapult.Pagination `json:"pagination,omitempty"`
+	Tags       []*Tag               `json:"tags,omitempty"`
+	Tag        *Tag                 `json:"tag,omitempty"`
+}
+
+func (s *TagsClient) List(
+	ctx context.Context,
+	ref OrganizationRef,
+	opts *ListOptions,
```

Comment:
Actually, to make this more annoying, the tags list endpoint does NOT support pagination... LOL 🤦‍♂️

https://developers.katapult.io/docs/api/latest/core/groups/tags/

Though, I'd say we should probably consider making it paginated. In theory I would say all list endpoints in the API should support pagination, possibly with exceptions for specific global resources which will only contain a handful of items.

So let's discuss about the way forward here before removing the pagination stuff as it's already here :)

Reply:
^ is this blocked on this ?

Reply:
We are leaving the pagination stuff in.